explorer-ui/src/components/Activities/Activities.jsx

```diff
@@ -74,118 +77,77 @@ const  handleTripSubmit= async(event) => {
 
   // This function load POI's list page to the left pane. It uses 1000 meters radius for objects search:
 
-const loadList = async() => {
-    const data = await apiGet(
+  function loadList() {
+    apiGet(
       "radius",
       `radius=1000&limit=${pageLength}&offset=${offset}&lon=${37.6173}&lat=${55.7558}&rate=2&format=json`
-    )
-      let list = document.getElementById("list");
-      list.innerHTML = "";
+    ).then(function (data) {
       console.log("data: ", data);
-      setEventData(data);
+      data.forEach(element => setEventData(eventData => [...eventData, element]));
       console.log("event data: ", eventData);
-      getXIP(eventData)
-      // data.forEach(item => list.appendChild(createListItem(item)));
-      let nextBtn = document.getElementById("next_button");
-      if (count < offset + pageLength) {
-        nextBtn.style.visibility = "hidden";
-      } else {
-        nextBtn.style.visibility = "visible";
-        nextBtn.innerText = `Next (${offset + pageLength} of ${count})`;
-      }
+    });
   }
 
+ function loadMoreActivities(event){
+  offset += pageLength;
```

Comment:
likely prefer setting offset in state as global vars should generally be constants

Reply:
changed to setOffset(offset += pageLength)